src/main/java/com/iotiq/application/controller/SellerProductController.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+@RestController
+@RequiredArgsConstructor
+@PreAuthorize("hasRole('ROLE_COMPANY')")
+@RequestMapping("/api/v1/seller/{sellerId}/products")
+public class SellerProductController {
+
+    private ProductService productService;
+    private SellerService sellerService;
+
+    @GetMapping
+    @PreAuthorize("hasAuthority(@ProductManagementAuth.VIEW)")
+    public PagedResponse<ProductResponse> getAll(@PathVariable UUID sellerId, ProductFilter filter, Sort sort) {
+        checkScope(sellerId);
+
+        filter.setSellerIds(List.of(sellerId));
+        Page<Product> page = productService.getAll(filter, sort);
+        List<ProductResponse> responseList = ModelMapperUtil.map(page.getContent(), ProductResponse.class);
+
+        return PagedResponseBuilder.createResponse(page, responseList);
+    }
+
+    @GetMapping("/csv-export")
+    @PreAuthorize("hasAuthority(@ProductManagementAuth.VIEW)")
+    public ResponseEntity<byte[]> export(@PathVariable UUID sellerId) throws IOException {
+        checkScope(sellerId);
+
+        byte[] csvBytes = productService.exportCSVFile(sellerId);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_OCTET_STREAM);
+        headers.setContentDispositionFormData("attachment", "products.csv");
+
+        return ResponseEntity.ok()
+                .headers(headers)
+                .body(csvBytes);
+    }
+
+    @PostMapping("/csv-upload")
+    public ResponseEntity<ProductCSVUploadResponse> uploadFile(@PathVariable UUID sellerId, @RequestParam("file") MultipartFile file) {
+        checkScope(sellerId);
+        return productService.importCSVFile(sellerId, file);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasAuthority(@ProductManagementAuth.CREATE)")
+    @ResponseStatus(HttpStatus.CREATED)
+    public ProductCreateResponse createProduct(@PathVariable UUID sellerId, @RequestBody @Valid ProductCreateRequest request) {
+        checkScope(sellerId);
+        return new ProductCreateResponse(productService.createProduct(request).getId());
+    }
+
+    @DeleteMapping("/{id}")
+    @PreAuthorize("hasAuthority(@ProductManagementAuth.DELETE)")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void delete(@PathVariable UUID sellerId, @PathVariable("id") UUID id) {
+        checkScope(sellerId);
+        productService.delete(id);
```

Comment:
Not related to this pr only this is note: if product belongs at least one order, may be we should not allow to delete. 
Because it may cause data inconsistencies that we are not aware of or may not be able to detect in the future.
So may be we allow only mark as "discontinued" or "sold out". 🤔